In [1]:
import pandas as pd

In [3]:
df=pd.read_csv("Downloads/sales_customer_analytics.csv")

In [4]:
df.head()

,OrderID,OrderDate,CustomerID,CustomerName,City,Product,Category,Quantity,UnitPrice,DiscountPct,PaymentMode,OrderStatus,Rating
0,1001,2024-01-01,252,Pooja,Delhi,Laptop,Electronics,1,5852,0.10,UPI,Returned,5.0
1,1002,2024-01-01,293,Siddharth,Mumbai,Mobile,Electronics,1,27922,0.05,Cash,Delivered,5.0
2,1003,2024-01-01,215,Megha,Hyderabad,Mobile,Electronics,3,42376,0.15,Debit Card,Cancelled,5.0
3,1004,2024-01-01,272,Ananya,Hyderabad,Monitor,Electronics,1,61492,0.20,UPI,Delivered,3.0
4,1005,2024-01-02,261,Rohit,Chennai,Tablet,Electronics,1,18306,0.15,Cash,Cancelled,4.0


In [6]:
df.tail()

,OrderID,OrderDate,CustomerID,CustomerName,City,Product,Category,Quantity,UnitPrice,DiscountPct,PaymentMode,OrderStatus,Rating
545,1546,2024-06-28,284,Neha,Delhi,Monitor,Electronics,1,48761,0.15,Credit Card,Delivered,4.0
546,1547,2024-06-29,224,Pooja,Delhi,Desk,Furniture,3,10572,0.05,Debit Card,Delivered,5.0
547,1548,2024-06-29,212,Megha,Mumbai,Desk,Furniture,4,54018,NaN,UPI,Delivered,5.0
548,1549,2024-06-29,250,Manish,Chennai,Headphones,Accessories,4,45441,0.10,Credit Card,Cancelled,5.0
549,1550,2024-06-30,235,Megha,Mumbai,Monitor,Electronics,1,28342,NaN,Cash,Delivered,3.0


In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 550 entries, 0 to 549
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   OrderID       550 non-null    int64  
 1   OrderDate     550 non-null    object 
 2   CustomerID    550 non-null    int64  
 3   CustomerName  550 non-null    object 
 4   City          550 non-null    object 
 5   Product       550 non-null    object 
 6   Category      550 non-null    object 
 7   Quantity      550 non-null    int64  
 8   UnitPrice     550 non-null    int64  
 9   DiscountPct   458 non-null    float64
 10  PaymentMode   550 non-null    object 
 11  OrderStatus   550 non-null    object 
 12  Rating        512 non-null    float64
dtypes: float64(2), int64(4), object(7)
memory usage: 56.0+ KB


In [8]:
df.describe()

,OrderID,CustomerID,Quantity,UnitPrice,DiscountPct,Rating
count,550.000000,550.000000,550.000000,550.000000,458.000000,512.000000
mean,1275.500000,250.258182,2.454545,33071.741818,0.099891,3.742188
std,158.915596,29.763651,1.108309,18322.527529,0.069894,1.139891
min,1001.000000,201.000000,1.000000,1007.000000,0.000000,1.000000
25%,1138.250000,224.000000,1.000000,17455.750000,0.050000,3.000000
50%,1275.500000,251.000000,2.000000,33185.000000,0.100000,4.000000
75%,1412.750000,275.000000,3.000000,49023.500000,0.150000,5.000000
max,1550.000000,300.000000,4.000000,64910.000000,0.200000,5.000000


In [15]:
df.describe(include="all")

,OrderID,OrderDate,CustomerID,CustomerName,City,Product,Category,Quantity,UnitPrice,DiscountPct,PaymentMode,OrderStatus,Rating
count,550.000000,550,550.000000,550,550,550,550,550.000000,550.000000,458.000000,550,550,512.000000
unique,NaN,182,NaN,20,6,9,3,NaN,NaN,NaN,4,3,NaN
top,NaN,2024-01-01,NaN,Pooja,Chennai,Chair,Electronics,NaN,NaN,NaN,UPI,Delivered,NaN
freq,NaN,4,NaN,37,102,68,243,NaN,NaN,NaN,156,457,NaN
mean,1275.500000,NaN,250.258182,NaN,NaN,NaN,NaN,2.454545,33071.741818,0.099891,NaN,NaN,3.742188
std,158.915596,NaN,29.763651,NaN,NaN,NaN,NaN,1.108309,18322.527529,0.069894,NaN,NaN,1.139891
min,1001.000000,NaN,201.000000,NaN,NaN,NaN,NaN,1.000000,1007.000000,0.000000,NaN,NaN,1.000000
25%,1138.250000,NaN,224.000000,NaN,NaN,NaN,NaN,1.000000,17455.750000,0.050000,NaN,NaN,3.000000
50%,1275.500000,NaN,251.000000,NaN,NaN,NaN,NaN,2.000000,33185.000000,0.100000,NaN,NaN,4.000000
75%,1412.750000,NaN,275.000000,NaN,NaN,NaN,NaN,3.000000,49023.500000,0.150000,NaN,NaN,5.000000


## 1.OrderID:-##
-It have 550 values, there are no missing values.
-Mean value 1275.50 is greater than Standard deviation value 158.915596, less data variated.
-1138.25 this value indicates that 25% of data is lie below this value.
-1275.50 this value indicates that 50% of data is lie below this value.
-1412.75 this value indicates that 75% of data is lie below this value.
-1550 is maximum value and 1001 is minimum value.
## 2.OrderDate:-##
-It have 550 values, there are no missing values.
-It having 182 unique values
-The most frequent value is 2024-01-01 because it is frequently appeared 4 times. 
## 3.CustomerID:-##
-It having 550 values.
-Mean value 250.258182 is greater than standard deviation value 29.763651 , less data variated.
-224 this value indicates that 25% of data is lie below this value.
-251 this value indicates that 50% of data is lie below this value.
-275 this value indicates that 75% of data is lie below this value.
-300 is maximum value and 201 is minimum value.
## 4.CustomerName:-##
-It have 550 values, there are no missing values.
-It having 20 unique values
-The most frequent value is Pooja because it is frequently appeared 37 times. 
## 5.City:-##
-It have 550 values, there are no missing values.
-It having 6 unique values
-The most frequent value is Chennai because it is frequently appeared 102 times. 
## 6.Product:-##
-It have 550 values, there are no missing values.
-It having 9 unique values
-The most frequent value is Chair because it is frequently appeared 68 times. 
## 7.Category:-##
-It have 550 values, there are no missing values.
-It having 3 unique values
-The most frequent value is Electronics because it is frequently appeared 243 times. 
## 8.Quantity:-##
-It having 550 values.
-Mean value 2.454545 is greater than standard deviation value 1.108309 , less data variated.
-1 this value indicates that 25% of data is lie below this value.
-2 this value indicates that 50% of data is lie below this value.
-3 this value indicates that 75% of data is lie below this value.
-4 is maximum value and 1 is minimum value.
## 9.UnitPrice:-##
-It having 550 values.
-Mean value 33071.741818 is greater than standard deviation value 18322.527529 , less data variated.
-17455.75 this value indicates that 25% of data is lie below this value.
-33185 this value indicates that 50% of data is lie below this value.
-49023.50 this value indicates that 75% of data is lie below this value.
-64910 is maximum value and 1007 is minimum value.
## 10.DiscountPct:-##
-It having 458 values, there are 92 missing values.
-Mean value 0.099891 is greater than standard deviation value 0.069894 , less data variated.
-0.050 this value indicates that 25% of data is lie below this value.
-0.10 this value indicates that 50% of data is lie below this value.
-0.15 this value indicates that 75% of data is lie below this value.
-0.20 is maximum value and 0 is minimum value.
## 11.PaymentMode:-##
-It have 550 values, there are no missing values.
-It having 4 unique values
-The most frequent value is UPI because it is frequently appeared 156 times. 
## 12.OrderStatus:-##
-It have 550 values, there are no missing values.
-It having 3 unique values
-The most frequent value is Delivered because it is frequently appeared 457 times. 
## 13.Rating:-##
-It having 512 values, there are 38 missing values.
-Mean value 3.742188 is greater than standard deviation value 1.139891 , less data variated.
-0.050 this value indicates that 25% of data is lie below this value.
-0.10 this value indicates that 50% of data is lie below this value.
-0.15 this value indicates that 75% of data is lie below this value.
-0.20 is maximum value and 0 is minimum value.


In [9]:
df.isna()

,OrderID,OrderDate,CustomerID,CustomerName,City,Product,Category,Quantity,UnitPrice,DiscountPct,PaymentMode,OrderStatus,Rating
0,False,False,False,False,False,False,False,False,False,False,False,False,False
1,False,False,False,False,False,False,False,False,False,False,False,False,False
2,False,False,False,False,False,False,False,False,False,False,False,False,False
3,False,False,False,False,False,False,False,False,False,False,False,False,False
4,False,False,False,False,False,False,False,False,False,False,False,False,False
...,...,...,...,...,...,...,...,...,...,...,...,...,...
545,False,False,False,False,False,False,False,False,False,False,False,False,False
546,False,False,False,False,False,False,False,False,False,False,False,False,False
547,False,False,False,False,False,False,False,False,False,True,False,False,False
548,False,False,False,False,False,False,False,False,False,False,False,False,False


In [10]:
df.isna().sum()

OrderID          0
OrderDate        0
CustomerID       0
CustomerName     0
City             0
Product          0
Category         0
Quantity         0
UnitPrice        0
DiscountPct     92
PaymentMode      0
OrderStatus      0
Rating          38
dtype: int64

In [11]:
df.duplicated().sum()

np.int64(0)

In [14]:
df.dtypes

OrderID           int64
OrderDate        object
CustomerID        int64
CustomerName     object
City             object
Product          object
Category         object
Quantity          int64
UnitPrice         int64
DiscountPct     float64
PaymentMode      object
OrderStatus      object
Rating          float64
dtype: object